public/scripts/main.js

```diff
@@ -189,81 +183,197 @@ const tooltip = new Tooltip({
 let readyToInteract = false;
 let inFightScene = false;
 let atEndPoint = false;
+
+// Tooltip for fight scene
+const skillA1Tooltip = new Tooltip({
+  x: 300,
+  y: 400,
+  width: 100,
+  height: 50,
+  text: 'A1 Z',
+});
+/*
+const skillA2Tooltip = new Tooltip({
+  x: 450,
+  y: 400,
+  width: 100,
+  height: 50,
+  text: 'A2 X',
+});
+
+const skillA3Tooltip = new Tooltip({
+  x: 600,
+  y: 400,
+  width: 100,
+  height: 50,
+  text: 'A3 C',
+});
+
+const skillA4Tooltip = new Tooltip({
+  x: 750,
+  y: 400,
+  width: 100,
+  height: 50,
+  text: 'A4 V',
+});
+
+const PlayerTooltip = new Tooltip({
+  x: 20,
+  y: 130,
+  width: 150,
+  height: 300,
+  text: playerStatText, // 'Bobby here! \nsmash all ppl \nblocking your way' + `${player.hp}`,
+});
+
+const EnemyTooltip = new Tooltip({
+  x: 820,
+  y: 30,
+  width: 150,
+  height: 300,
+  text: enemyStatText,
+});
+*/
+const fightTooltip = new Tooltip({
+  x: 250,
+  y: 30,
+  width: 500,
+  height: 150,
+  text: 'TIME TO FIGHT! coming later...\nE/SPACE TO RETURN',
+});
+
+const escapeTooltip = new Tooltip({
+  x: 300,
+  y: 320,
+  width: 550,
+  height: 150,
+  text: 'Q to normal attack; E to escape fight',
+});
+
+function fightSceneLoad(player, npc) {
+  const playerStatText = 'Bobby here! \nsmash all ppl \nblocking your way\n' + player.hp; //
+  const enemyStatText = 'Enemy: \ncome fight bobby\n\n' + npc.hp;
+  console.log(playerStatText);
+  const playerTooltip = new Tooltip({
+    x: 20,
+    y: 130,
+    width: 150,
+    height: 300,
+    text: playerStatText,
+  });
+
+  const enemyTooltip = new Tooltip({
+    x: 820,
+    y: 30,
+    width: 150,
+    height: 300,
+    text: enemyStatText,
+  });
+  fightLayer.add(skillA1Tooltip.renderBox, skillA1Tooltip.renderText);
+  /*
+  fightLayer.add(skillA2Tooltip.renderBox, skillA2Tooltip.renderText);
+  fightLayer.add(skillA3Tooltip.renderBox, skillA3Tooltip.renderText);
+  fightLayer.add(skillA4Tooltip.renderBox, skillA4Tooltip.renderText);
+  */
+  fightLayer.add(playerTooltip.renderBox, playerTooltip.renderText);
+  fightLayer.add(enemyTooltip.renderBox, enemyTooltip.renderText);
+  fightLayer.add(fightTooltip.renderBox, fightTooltip.renderText);
+  fightLayer.add(escapeTooltip.renderBox, escapeTooltip.renderText);
+  fightLayer.draw();
+}
+
 // Handles keys being pressed down
 container.addEventListener('keydown', function(event) {
   keys[event.keyCode] = true;
 
-  doKeyProcess(keys);
```

Comment:
Added login branch of keys so if in fight, run the fightloop function that trigger another set of key.